hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -154,6 +154,7 @@ const (
 
 	previouslySyncedRestartDateAnnotation = "hypershift.openshift.io/previous-restart-date"
 	kasServingCertHashAnnotation          = "hypershift.openshift.io/kas-serving-cert-hash"
+	referencedResourceAnnotationPrefix    = "hypershift.openshift.io/referenced-resource-"
```

Comment:
My pref would be something like:
`referenced-resource.hypershift.openshift.io/[name]`
A little less confusing when understanding what part is prefix and what part is name.